### PR TITLE
[ShizaProject] Implement auto re-auth on cookie expire

### PR DESCRIPTION
**Pull request by dm3ch**
Fixes https://github.com/Jackett/Jackett/pull/10166#issuecomment-739589464
Fixes https://github.com/Jackett/Jackett/issues/10160

### CODE REVIEW

**Comment by dm3ch** [comment]
I tested that 2 different ways:
1) I tried to test my code on my own old config. But instead of successful re-auth Jacket tried to send `|||%%PREVJACKPAS` instead of  actual password.
Suddenly I haven't found the reason of that.

2) After that I reconfigured indexer. After that I have manually corrupted cookie in json multiple times with restart after that. This times re-auth worked as expected.

File: src/Jackett.Common/Indexers/ShizaProject.cs [hunk 1]
```diff
@@ -57,7 +58,7 @@ public ShizaProject(IIndexerConfigurationService configService, WebClient wc, Lo
         private string SearchUrl => SiteLink + "releases/search";
 
         public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken configJson)
-        {
+        {
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/ShizaProject.cs:61]: clean up

File: src/Jackett.Common/Indexers/ShizaProject.cs [hunk 2]
```diff
@@ -13,6 +13,7 @@
 using Jackett.Common.Utils.Clients;
 using Newtonsoft.Json.Linq;
 using NLog;
+using Newtonsoft.Json;
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/ShizaProject.cs:16]: unused import

File: src/Jackett.Common/Indexers/ShizaProject.cs [hunk 3]
```diff
@@ -193,7 +195,13 @@ private async Task<List<ReleaseInfo>> FetchShowReleases(string url)
         private bool IsAuthorized(WebResult result) {
             var parser = new HtmlParser();
             var document = parser.ParseDocument(result.ContentString);
-            return document.QuerySelector("div.profile-menu > a").Attributes["href"].Value.EndsWith("/logout");
+            try {
+                return document.QuerySelector("div.profile-menu > a").Attributes["href"].Value.EndsWith("/logout");
```
Comment (ngosang) [inline src/Jackett.Common/Indexers/ShizaProject.cs:199]: IMHO you can simplify the code. Just string.Contains is enough and less error prone.
https://github.com/Jackett/Jackett/blob/2ea22af9efeb2ec85c2cfb93c9882f8a616843fb/src/Jackett.Common/Indexers/Nebulance.cs#L102

**Comment by dm3ch** [comment]
@ngosang Fixed review issues. Thank you for idea of more easy IsAuthorised check.
I tested that and seems with this approach all still works fine.

I also have removed passing cookies to login request, cause with them site responds with just redirect that doesn't contains login error text.

**Review by ilike2burnthing — APPROVED** [review]
Yep, that's working fine. Nicely done.

**Review by ngosang — APPROVED** [review]

**Comment by ngosang** [comment]
Thank you!

**Pull request merged**